src/java/org/apache/cassandra/service/reads/ReadCallback.java

```diff
@@ -95,12 +93,15 @@ protected P replicaPlan()
         return replicaPlan.get();
     }
 
-    public boolean await(long timePastStart, TimeUnit unit)
+    /**
+     * await until the condition is signalled but not longer than timePastStart since startTimeNanos
+     */
+    public boolean awaitFrom(long startTimeNanos, long timePastStart, TimeUnit unit)
```

Comment:
The original `await` name was confusing. It's typically associated with waiting a specified amount of time, which was not the case in the original code.